app/notification.go

```diff
@@ -434,7 +434,10 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 			}
 			if sendEvent {
 				message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_THREAD_UPDATED, team.Id, "", uid, nil)
-				userThread, _ := a.Srv().Store.Thread().GetThreadForUser(uid, channel.TeamId, thread.PostId, true)
+				userThread, err := a.Srv().Store.Thread().GetThreadForUser(uid, channel.TeamId, thread.PostId, true)
+				if err != nil {
+					return nil, errors.Wrapf(err, "cannot get thread %q for user %q", thread.PostId, uid)
+				}
 				a.sanitizeProfiles(userThread.Participants, false)
```

Comment:
There also seems to be a case where it returns `nil, nil`

```
if !thread.Following {
	return nil, nil // in case the thread is not followed anymore - return nil error to be interpreted as 404
}
```

Should we be handling that?

Reply:
Ouch. Most definitely. I implicitly thought it was respecting the common pattern. Nice catch :+1:

Reply:
@reflog Do you want this case handled as an error or should we simply pass through?

Reply:
@streamer45 - no, this is an actual bug. If user is not following the thread, we should send nothing. I'll add a fix to your PR if you don't mind